Can't use FindConditions with generics

## Issue Description

### Expected Behavior

While coding abstract classes for my application, I would like to use generic types as their entities and be able to execute a query.


### Actual Behavior

I've created an abstract class that is extended by my entity services. While doing so, I have some methods that can execute queries. Everything is working as expected, but my type system breaks as soon as I use any method that uses `FindCondition` as an argument. Here is a minimal example:

```typescript
abstract class GraphQLEntityResolver<E extends DomainEntity, D> {
  
  (...)

  public async getUserPolicies(selector: FindConditions<E>, user: AuthzUser) {
    const actionSelectors: AuthzScopeGroup = {
      [ACTION.CREATE]: user.scopes[this.resource][ACTION.CREATE],
      [ACTION.READ]: user.scopes[this.resource][ACTION.READ],
      [ACTION.UPDATE]: user.scopes[this.resource][ACTION.UPDATE],
      [ACTION.DELETE]: user.scopes[this.resource][ACTION.DELETE],
    }

    const policies: ActionPolicies = mapValues(
      actionSelectors,
      async (constraint, action: ACTION): Promise<POLICY> => {
        if (!constraint) return POLICY.DENY
        const foundData = await this.getOneWithActionScopeConstraint(selector, user, action)

        return foundData ? POLICY.ALLOW : POLICY.DENY
      },
    )

    return policies
  }

  @ResolveField('policies', () => PolicyObject)
  protected async getEntityPolicies(
    @Parent() entity: EntityObject,
    @GraphQLUser() user: AuthzUser,
  ) {
    const selector: FindConditions<E> = { id: entity.id }

    return this.getUserPolicies(selector, user)
  }
}
```

At the `getEntityPolicies` method, the selector I'm building is giving me the following error:
```
src/app/graphql/resolver.ts:138:11 - error TS2322: Type '{ id: string; }' is not assignable to type 'FindConditions<E>'.
```

But, here is the `DomainEntity` entity that I'm using:
```typescript
export abstract class DomainEntity {
  @PrimaryGeneratedColumn('uuid')
  public id: string

  @CreateDateColumn()
  public createdAt: Date
}
```

Also, if I change my class to:
```typescript
abstract class GraphQLEntityResolver<D> {
```

And always use `DomainEntity` instead of `E`, it works as expected.

So, it seems that `FindConditions<E>` is not considering the keys of the `DomainEntity` abstract entity class. Since it has `id` but the selector is complaining, saying that I can't use ID as a selector.

--- a/src/find-options/FindConditions.ts
+++ b/src/find-options/FindConditions.ts
@@ -4,5 +4,5 @@ import {FindOperator} from "./FindOperator";
  * Used for find operations.
  */
 export type FindConditions<T> = {
-    [P in keyof T]?: T[P] extends never ? FindConditions<T[P]>|FindOperator<FindConditions<T[P]>> : FindConditions<T[P]>|FindOperator<FindConditions<T[P]>>;
+    [P in keyof T]?: FindConditions<T[P]> | FindOperator<FindConditions<T[P]>>;
 };

